src/dataflow/source/kafka.rs

```diff
@@ -345,6 +345,11 @@ fn downgrade_capability(
                                    to Kafka sources."
                 ),
             };
+            if pid as usize > partition_metadata.len() {
```

Comment:
pids in Kafka start at 0, so pid 1 should trigger the partition_metadata to be extended, no?

Reply:
yep, sorry, should have tested more before putting this up!